ballista/rust/core/src/execution_plans/shuffle_reader.rs

```diff
@@ -102,29 +107,35 @@ impl ExecutionPlan for ShuffleReaderExec {
     fn execute(
         &self,
         partition: usize,
-        _context: Arc<TaskContext>,
+        context: Arc<TaskContext>,
     ) -> Result<SendableRecordBatchStream> {
-        debug!("ShuffleReaderExec::execute({})", partition);
-
-        let fetch_time =
-            MetricBuilder::new(&self.metrics).subset_time("fetch_time", partition);
-
-        let locations = self.partition[partition].clone();
-        let stream = locations.into_iter().map(move |p| {
-            let fetch_time = fetch_time.clone();
-            futures::stream::once(async move {
-                let timer = fetch_time.timer();
-                let r = fetch_partition(&p).await;
-                timer.done();
-
-                r.map_err(|e| ArrowError::ExternalError(Box::new(e)))
-            })
-            .try_flatten()
-        });
+        let task_id = context.task_id().unwrap_or_else(|| partition.to_string());
+        info!("ShuffleReaderExec::execute({})", task_id);
+
+        // TODO make the maximum size configurable, or make it depends on global memory control
+        let max_request_num = 50usize;
```

Comment:
+1 for making this a config option that is documented in the user guide. I can help with this in a follow-on PR.